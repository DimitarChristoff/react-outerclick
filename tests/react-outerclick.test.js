import React      from 'react'
import renderer   from 'react-test-renderer'
import outerClick from '../index'

jest.mock('react-dom')

describe('Outerclick tests >', () => {

  it('renders wrapped components correctly', () => {
    class C extends React.Component {
      render(){
        return <div>{this.props.message}</div>
      }
    }

    const Component = outerClick(C);

    const tree = renderer.create(<Component message="Hi" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

})
