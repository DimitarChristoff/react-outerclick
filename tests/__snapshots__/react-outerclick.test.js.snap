exports[`Outerclick tests > renders wrapped components correctly 1`] = `
<div>
  Hi
</div>
`;
